sklearn/preprocessing/data.py

```diff
@@ -2873,3 +2875,32 @@ def inverse_transform(self, X):
                     X_tr[mask, idx] = None
 
         return X_tr
+
+    def get_feature_names(self, input_features=None):
+        """Return feature names for output features
+
+        Parameters
+        ----------
+        input_features : list of string, length n_features, optional
+            String names for input features if available. By default,
+            "x0", "x1", ... "xn_features" is used.
+
+        Returns
+        -------
+        output_feature_names : list of string, length n_output_features
+
+        """
+        cats = self.categories_
+        feature_names = []
+        if input_features is None:
+            input_features = ['x%d' % i for i in range(len(cats))]
+
+        for i in range(len(cats)):
+            temp = []
+            for t in cats[i]:
+                temp.append(str(t))
+            t = [input_features[i] + '_' + f for f in temp]
+            feature_names.append(t)
```

Comment:
why not just

```
    feature_names.extend(input_features[i] + '_' + str(t)
                         for t in cats[i])
```

Reply:
I also think we should be using unicode here in python 2... but I'm not sure.

Reply:
I've added the 2 changes you mentioned. Will make a PR soon.I also tried with encoding = 'ordinal' and it worked with that too.

Reply:
I don't understand what you mean by `using unicode here` . Can you explain that better?

Reply:
No need to make a new PR, just push more commits to this branch, please.

Reply:
In Python 2, we need to use `unicode` not `str`, etc, to allow for non-ascii strings.

Reply:
@jnothman I am not sure this conversion to unicode is needed. 
If a user has unicode categories or passes unicode input features, concatenating them still works as expected in python 2 (a mixture of string and unicode will just automatically be unicode as result). 
Not doing this conversion AFAIK only means we can end up with a list of both string and unicode values in python 2.

Reply:
OK, will do some more testing on this (I actually expect the current tests to fail on python 2, let's see)